rezz.lic

```diff
@@ -80,14 +80,28 @@ class Rezz
     @vigil ||= { 'abbrev' => 'vigil', 'prep_time' => 10, 'cast' => "cast #{player}", 'mana' => 20 }
     vigil(player) unless cast_spell(@vigil, @settings)
   end
+  
+  def infuse_amount
+    if DRStats.mana < 20
+      return (@rezz['mana'] / 5)
+    elsif DRStats.mana < 40
+      return (@rezz['mana'] / 4)
+    elsif DRStats.mana < 61
+      return (@rezz['mana'] / 3)
+    else
+      return @rezz['mana']
+    end
+  end
 
   def find_soul?(player)
     Flags.add('rotting', /#{player}.s body'?s? (grows paler|seems to|appears to|appearance takes)/)
     15.times do
-      pause 5 while DRStats.mana < 40
-      @settings.osrel_no_harness ? nil : harness_mana([@rezz['mana']])
-      match = bput("infuse rezz #{@rezz['mana']}", 'You sense a spirit nearby, but you are unable to make it out clearly', /spirit of #{player} in the Void/)
+      infuse_mana = infuse_amount
+      pause 5 while DRStats.mana < 20
+      @settings.osrel_no_harness ? nil : harness_mana([infuse_mana])
+      match = bput("infuse rezz #{infuse_mana}", 'You sense a spirit nearby, but you are unable to make it out clearly', /spirit of #{player} in the Void/)
       @infuse_attempts += 1
```

Comment:
So, you have this code:
```
    until find_soul?(player)
      unless keep_going?	      unless keep_going?
        bput("whisper #{player} I can't find your soul.  You need more favors!", 'You whisper', 'Who are you trying to whisper to?')	        bput("whisper #{player} I can't find your soul.  You need more favors!", 'You whisper', 'Who are you trying to whisper to?')
        return	        return
      end	      end
    end	    end
```
You ask every infuse for the player to put `;send yes/no`? This seems less automated than I would personally expect. Why not give the player the ability to set the attempts they want before asking? Or maybe, ask if under 30% mana only, etc. Something like that? Every iteration seems a bit excessive. Nothing to do with this PR, but I wanted to make note of it seeing how the attempts are still being used.